packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -106,28 +105,43 @@ class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
     accessibility: treeItemBehavior,
   }
 
-  titleRef = React.createRef<HTMLElement>()
+  itemRef = React.createRef<HTMLElement>()
   treeRef = React.createRef<HTMLElement>()
 
   actionHandlers = {
-    getFocusFromParent: e => {
-      const { open } = this.props
-      if (open) {
+    performClick: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      _.invoke(this.props, 'onTitleClick', e, this.props)
+    },
+    collapseOrReceiveFocus: e => {
```

Comment:
may be even shorter: `collapseOrFocus`

Reply:
general thoughts on current `actionHandlers` API. As a client, frankly, it is completely unobvious for me how I should respond on `collapseOrFocus` action - specifically, what drives my confusion is this `or` part. What is the principle based on which I should select one or the other? It turns out that it is based on `open` prop value - but how I would know that? 

Ok, and then the question is - I have provided all the props to accessibility behavior, so I might expect that it has access to all of them. Then, why it doesn't just expose `collapse` or `focus` actions, because I expect that inner implementation might just check `open` prop I've provided to it - and then I, as a client, won't be required to solve this puzzle on my side.

Really think that we should adjust implementation of action handlers a bit, to make this decision for the client (not this PR - just general thoughts). This will result in much more convenient API

Reply:
@jurokapsiar @sophieH29 @kolaps33 We can follow up.

Reply:
agree, action handlers can be extended to reflect this kind of situation - let's discuss how the API would look like

Reply:
https://github.com/stardust-ui/react/issues/1502